@copilot fix

test/UnitTests/MSTest.Analyzers.UnitTests/Verifiers/CSharpCodeFixVerifier`2.cs#L92

test/UnitTests/MSTest.Analyzers.UnitTests/Verifiers/CSharpCodeFixVerifier`2.cs(92,1): error : WhenAssertionActionIsTargetTypedDelegateCreation_CodeFixUnwrapsDelegateCreation [net8.0]
Test method MSTest.Analyzers.Test.PreferAsyncAssertionAnalyzerTests.WhenAssertionActionIsTargetTypedDelegateCreation_CodeFixUnwrapsDelegateCreation threw exception:
System.InvalidOperationException: Context: Diagnostics of test state
Mismatch between number of diagnostics returned, expected "1" actual "3"

Diagnostics:
// /0/Test0.cs(11,9): info MSTEST0064: Use 'Assert.ThrowsExactlyAsync' instead of blocking an async call in 'Assert.ThrowsExactly'
VerifyCS.Diagnostic().WithSpan(11, 9, 11, 104).WithArguments("ThrowsExactlyAsync", "ThrowsExactly"),
// /0/Test0.cs(11,67): error CS0029: Cannot implicitly convert type 'void' to 'object'
DiagnosticResult.CompilerError("CS0029").WithSpan(11, 67, 11, 102).WithArguments("void", "object"),
// /0/Test0.cs(11,67): error CS1662: Cannot convert lambda expression to intended delegate type because some of the return types in the block are not implicitly convertible to the delegate return type
DiagnosticResult.CompilerError("CS1662").WithSpan(11, 67, 11, 102).WithArguments("lambda expression"),

 [Check failure on line 1067 in test/UnitTests/MSTest.Analyzers.UnitTests/PreferAsyncAssertionAnalyzerTests.cs](https://github.com/microsoft/testfx/pull/8329/files#annotation_55924448647) 

@azure-pipelines
azure-pipelines
/ microsoft.testfx (Build Linux Debug)
test/UnitTests/MSTest.Analyzers.UnitTests/PreferAsyncAssertionAnalyzerTests.cs#L1067

test/UnitTests/MSTest.Analyzers.UnitTests/PreferAsyncAssertionAnalyzerTests.cs(1067,1): error : WhenNonAsyncTaskMethodHasReturnInsideFixedBlock_DiagnosticReportedButNoCodeFixOffered [net8.0]
Test method MSTest.Analyzers.Test.PreferAsyncAssertionAnalyzerTests.WhenNonAsyncTaskMethodHasReturnInsideFixedBlock_DiagnosticReportedButNoCodeFixOffered threw exception:
System.InvalidOperationException: Context: Diagnostics of test state
Mismatch between number of diagnostics returned, expected "1" actual "0"

Diagnostics:
    NONE.

--- a/test/UnitTests/MSTest.Analyzers.UnitTests/PreferAsyncAssertionAnalyzerTests.cs
+++ b/test/UnitTests/MSTest.Analyzers.UnitTests/PreferAsyncAssertionAnalyzerTests.cs
@@ -919,7 +919,7 @@ public sealed class PreferAsyncAssertionAnalyzerTests
                 [TestMethod]
                 public void MyTestMethod()
                 {
-                    [|Assert.ThrowsExactly<InvalidOperationException>(new((Action)(() => BarAsync().GetAwaiter().GetResult())))|];
+                    [|Assert.ThrowsExactly<InvalidOperationException>((Action)new(() => BarAsync().GetAwaiter().GetResult()))|];
                 }
 
                 private Task BarAsync() => Task.CompletedTask;

--- a/test/UnitTests/MSTest.Analyzers.UnitTests/PreferAsyncAssertionAnalyzerTests.cs
+++ b/test/UnitTests/MSTest.Analyzers.UnitTests/PreferAsyncAssertionAnalyzerTests.cs
@@ -919,7 +919,7 @@ public sealed class PreferAsyncAssertionAnalyzerTests
                 [TestMethod]
                 public void MyTestMethod()
                 {
-                    [|Assert.ThrowsExactly<InvalidOperationException>(new(() => BarAsync().GetAwaiter().GetResult()))|];
+                    [|Assert.ThrowsExactly<InvalidOperationException>(new((Action)(() => BarAsync().GetAwaiter().GetResult())))|];
                 }
 
                 private Task BarAsync() => Task.CompletedTask;
@@ -1035,13 +1035,16 @@ public sealed class PreferAsyncAssertionAnalyzerTests
             public class MyTestClass
             {
                 [TestMethod]
-                public unsafe Task MyTestMethod()
+                public Task MyTestMethod()
                 {
                     [|Assert.ThrowsExactly<InvalidOperationException>(() => BarAsync().GetAwaiter().GetResult())|];
                     int[] data = { 1, 2, 3 };
-                    fixed (int* p = data)
+                    unsafe
                     {
-                        return Task.CompletedTask;
+                        fixed (int* p = data)
+                        {
+                            return Task.CompletedTask;
+                        }
                     }
                 }
 

